### PR TITLE
Docs: Update terminology

### PR DESCRIPTION
**Pull request by shati-patel**
A more in-depth attempt at changing terminology for GHU. I've only updated the non language-specific topics so far. ~I'll probably raise separate PRs for each language.~ I'll add separate commits for each language.

Some issues/questions I had: 
- QL training: I haven't changed this section yet, but the slides talk about specific snapshots of a project (before the vulnerabilities are fixed). This might be a bit more subtle than just replacing "snapshot" with "database".
- There are lots of links to LGTM/QL for Eclipse/QL command-line tools. I've left most of these in place, but presumably they should go to new links in future? Similarly for links to the [LGTM security page](https://lgtm.com/security) or to [Semmle's variant analysis page](https://semmle.com/variant-analysis).
- There's one "new" page in this PR - this is just from renaming `snapshot.rst` to `database.rst`. I _think_ I've updated all the links to this page...
- In most places, I've changed "Semmle repository" to "GitHub repository". Perhaps this isn't necessary (yet), but it seemed to work in most places.

Anyway, could you have a look at the changes so far, and see if this is the kind of thing we'd want to do?

### CODE REVIEW

File: docs/language/learn-ql/introduction-to-ql.rst [hunk 1]
```diff
@@ -1,18 +1,22 @@
-Introduction to the QL language
-===============================
+Introduction to QL 
+==================
 
-QL is a powerful query language that is used to analyze code. Queries written in QL can be used to find errors and uncover variants of important security vulnerabilities. Visit Semmle's `security research page <https://lgtm.com/security>`__ to read about examples of vulnerabilities that we have recently found in open source projects using QL queries.
+QL is a powerful query language that underlies CodeQL, which is used to analyze code.
```
Comment (felicitymay) [inline docs/language/learn-ql/introduction-to-ql.rst]:
```suggestion
QL is a powerful query language that underlies CodeQL, which is used to analyze code.```

**Comment by shati-patel** [comment]
Thanks for having a look! I've made a few small changes to remove CodeQL in certain places and fix a few more mistakes.

> You're right to point out that those links that will need to be addressed in the future - can you create a Jira issue for them to ensure that we don't forget them? 

Done, see [SD-4016](https://jira.semmle.com/browse/SD-4016).

File: docs/language/learn-ql/introduction-to-ql.rst (hunk 1 above)
Comment (felicitymay) [reply docs/language/learn-ql/introduction-to-ql.rst]: No - sorry, looks like a typo on my side 😞

**Comment by felicitymay** [comment]
The COBOL and C/C++ commits look good too. Thanks.

File: docs/language/learn-ql/java/introduce-libraries-java.rst [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
+=========================================
 
 Overview
 --------
 
-There is an extensive QL library for analyzing Java code. The QL classes in this library present the data from a snapshot database in an object-oriented form and provide abstractions and predicates to help you with common analysis tasks.
+There is an extensive library for analyzing CodeQL databases extracted from Java projects. The classes in this library present the data from a database in an object-oriented form and provide abstractions and predicates to help you with common analysis tasks.
 
-The library is implemented as a set of QL modules, that is, files with the extension ``.qll``. The module ``java.qll`` imports all other standard library modules, so you can include the complete library by beginning your query with:
+The library is implemented as a set of QL modules, that is, files with the extension ``.qll``. The module ``java.qll`` imports all the core C/C++ library modules, so you can include the complete library by beginning your query with:
```
[the remark this answers is not in the document]
Comment (shati-patel) [reply docs/language/learn-ql/java/introduce-libraries-java.rst]: Woops! Good spot, thank you.

File: docs/language/learn-ql/python/introduce-libraries-python.rst [hunk 3]
```diff
@@ -1,39 +1,35 @@
...
    import python
 
-The rest of this tutorial summarizes the contents of the standard QL libraries. We recommend that you read this and then work through the practical examples in the Python tutorials shown at the end of the page.
+The rest of this tutorial summarizes the contents of the standard libraries for Python. We recommend that you read this and then work through the practical examples in the tutorials shown at the end of the page.
 
 Overview of the library
 -----------------------
 
-The QL Python library incorporates a large number of classes, each class corresponding either to one kind of entity in Python source code or to an entity that can be derived form the source code using static analysis. These classes can be divided into four categories:
+The CodeQL library for Python incorporates a large number of classes. Each class corresponds either to one kind of entity in Python source code or to an entity that can be derived form the source code using static analysis. These classes can be divided into four categories:
```
Comment (felicitymay) [inline docs/language/learn-ql/python/introduce-libraries-python.rst]: Typo in the original
```suggestion
The CodeQL library for Python incorporates a large number of classes. Each class corresponds either to one kind of entity in Python source code or to an entity that can be derived from the source code using static analysis. These classes can be divided into four categories:
```

File: docs/language/learn-ql/python/taint-tracking.rst [hunk 4]
```diff
@@ -222,7 +222,7 @@ The ``TaintTracking::Source`` and ``TaintTracking::Sink`` classes have predicate
         ...
     }
 
-The ``TaintKind`` itself is just a string (a QL string, not a QL entity representing a Python string),
+The ``TaintKind`` itself is just a string (a QL string, not a CodeQL entity representing a Python string),
```
Comment (felicitymay) [inline docs/language/learn-ql/python/taint-tracking.rst]: Hurrah! a place where the new terminology makes the text clearer 😄

File: docs/language/learn-ql/writing-queries/path-queries.rst [hunk 5]
```diff
@@ -103,7 +103,7 @@ Generating path explanations
...
 To do this you need to define a `query predicate <https://help.semmle.com/QL/ql-handbook/queries.html#query-predicates>`__ called ``edges`` in your query.
 This predicate defines the edge relations of the graph you are computing, and it is used to compute the paths related to each result that your query generates. 
-You can import a predefined ``edges`` predicate from a path graph module in one of the standard QL data flow libraries. In addition to the path graph module, the data flow libraries contain the other ``classes``, ``predicates``, and ``modules`` that are commonly used in data flow analysis. The import statement to use depends on the language that you are analyzing.
+You can import a predefined ``edges`` predicate from a path graph module in one of the standard data flow libraries. In addition to the path graph module, the data flow libraries contain the other ``classes``, ``predicates``, and ``modules`` that are commonly used in data flow analysis. The import statement to use depends on the language that you are analyzing.
 
 For C/C++, C#. Java, and JavaScript you would use::
```
Comment (shati-patel) [inline docs/language/learn-ql/writing-queries/path-queries.rst]:
```suggestion
For C/C++, C#, Java, and JavaScript you would use::
```

Reminder to self to fix this too ^

**Comment by shati-patel** [comment]
I think everything in the "Learning CodeQL" project is now internally consistent. 
(I updated the navigation bar in that latest commit.)

**Comment by felicitymay** [comment]
Thanks. That's great.

**Comment by jf205** [comment]
Thanks for doing this @shati-patel. 
@felicitymay, please go ahead and merge if you think it's ready.

**Pull request merged**